app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,97 @@
+
+
+class TasksController < ApplicationController
+  def index
+    @tasks = Task.all
+  end
+
+  def show
+    task_id = params[:id].to_i
+    @task = Task.find_by(id: task_id)
+
+
+    if @task.nil?
+      head 302
+      return
+    end
+  end
+
+  def new
+    @task = Task.new
+  end
+
+  def create
+    @task = Task.new(
+        name: params[:task][:name],
+        description: params[:task][:description],
+        completed_at: params[:task][:completed_at]
+    )
+    if @task.save
+      redirect_to task_path(@task.id)
+      return
+    else
+      render :new
+      return
+    end
+  end
+
+  def edit
+    @task = Task.find_by(id: params[:id])
+
+    if @task.nil?
+      redirect_to tasks_path
+      return
+    end
+  end
+
+  def update
+    @task = Task.find_by(id: params[:id])
+    task_params = params.require(:task).permit(:name, :description, :completed_at)
```

Comment:
This line is used in several methods and could be DRYed up by creating a private task_params method like we demonstrate with AdaBooks